### PR TITLE
Burner Heavy Assault Rifle (.50 cal)

### PR DESCRIPTION
**Pull request by a-person5660**
## About the PR
this branch started as my wish for higher caliber guns to feel more powerful and eventually turned into me making a .50 cal full auto rifle with explosive bullets lmao.

This PR adds the Burner heavy assault rifle, it fires a new caliber of bullet (.50), there are two types of ammunition for this caliber; anti-material and high-explosive. anti-material deals 45 damage and 160 structural damage, high explosive causes a small 3x3 explosion where it lands that deals aproximately the same damage as the anti material to the entity it directly hits and about half of that to the surrounding tiles, it is incapable of spacing tiles directly or ripping up floor/breaking wires but it can break walls and windows. its magazines are only 10+1 rounds big and the gun has a fire rate of 2.5, which is half of the lecter, it also has about the same TTK as the L6 Saw with both HE and AM rounds.

## Why / Balance
This is inteded to be a side-grade to the L6 Saw and China Lake, because of that i have placed this at 150 TC, like the L6 i have put two magazines for it in the ammo bundle.
To balance the explosive damage from the high explosive rounds you are only provided 1 spare mag and 1 spare HE mag in the bundle with the only way to obtain more is to buy an ammo bundle which contains another high explosive magazine.
Magazines for this gun are very hard to come by, the bundle comes with two spare magazines and one in the gun, an ammo bundle contains two magazines, with the intent that you will have to watch your ammo and time when to repack your magazines. Ammo boxes you can get from the Burner bundle, ammo bundle and purchasing straight from the uplink for 25 TC each, they contain 30 rounds each.

as for the .30 rifle buff, it just increases its damage to be more than 3 more than .20 rifle (AKMS enjoyers rejoyce!)

discussed this a lot in the discord and everyone generally aggrees that a sidegrade to the L6 with opposite streangths/weaknesses would be very cool.

## Media

image

https://github.com/user-attachments/assets/12df72e5-6ab9-46ec-9091-74918ed626da
(mildly outdated but you get the gyst) 

## big gun shoot big bullet, kill many crew!

**Changelog**

:cl:
- add: Added the Burner heavy assault rifle to the synicate and nuclear operative uplinks for 150 TC
- tweak: Ammo bundles now contain ammunition for the Burner heavy assault rifle

### CODE REVIEW

**Comment by a-person5660** [comment]
fuck, yaml fail, i swear it passed the linter on my device before i made the PR 😭

File: Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Rifles/rifle.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    sprite: _Goobstation\Objects\Weapons\Guns\Rifles\burner.rsi
+    layers:
+      - state: base
+        map: ["enum.GunVisualLayers.Base"]
+      - state: mag-0
+        map: ["enum.GunVisualLayers.Mag"]
+  - type: Gun
+    fireRate: 3
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/sniper.ogg
```
[the remark this answers is not in the document]
Comment (a-person5660) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Rifles/rifle.yml:17]: i meant to write 3, this is fine as is.

File: Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 2]
```diff
@@ -66,3 +66,47 @@
...
+  parent: BaseBullet
+  id: BulletHighCaliber
+  name: bullet (.50 anti-materiel)
+  components:
+  - type: Projectile
+    damage:
+      types:
+        Piercing: 45
+        Structural: 160 # if you were a wall would you like having a .50 cal anti materiel round go through you? (this breaks a normal wall in 3 shots, one mag goes through four normal walls, 10 rounds to take down a reinforced wall)
+
```
[the remark this answers is not in the document]
Comment (a-person5660) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:81]: i like muh structural, it is anti material for a reason

File: Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml [hunk 3]
```diff
@@ -67,6 +67,16 @@
...
+- type: listing
+  id: UplinkWeaponRifleBurner
+  name: uplink-weapon-burner-name
+  description: uplink-weapon-burner-desc
+  icon: { sprite: /Textures/_Goobstation\Objects\Weapons\Guns\Rifles\burner.rsi, state: icon }
+  productEntity: ClothingBackpackDuffelSyndicateBurner
+  cost:
+    Telecrystal: 150 # meant to be a side-grade to L6 Saw, if it ends up being better we can nerf it or up TC cost to 175-200.
+  categories:
+  - UplinkWeaponry
```
[the remark this answers is not in the document]
Comment (a-person5660) [reply Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml:79]: i need to flip all of the other slashes to match this one

**Comment by a-person5660** [comment]
me when minor spelling mistake

File: Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml [hunk 4]
```diff
@@ -226,3 +235,15 @@
...
+- type: listing
+  id: UplinkHighCaliberBox
+  name: uplink-high-caliber-box-name
+  description: uplink-high-caliber-box-desc
+  icon: { sprite: /Textures/Objects/Weapons/Guns/Ammunition/Boxes/anti_materiel.rsi, state: base-b }
+  productEntity: MagazineBoxHighCaliber
+  cost:
+    Telecrystal: 20 # may wanna change this number later
+  categories:
+  - UplinkAmmo
```
[the remark this answers is not in the document]
Comment (a-person5660) [reply Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml:249]: hristov is just way over priced, this gun also chews up ammo way quicker.

**Comment by a-person5660** [comment]
WTF IS THIS TEST FAIL WHYYYYY

**Comment by JoeHammad1844** [comment]
MFW full auto hristov

**Comment by JoeHammad1844** [comment]
Never mind I thought this used the same ammo as the hristov

**Comment by Piras314** [comment]
From my testing, it takes ~1 second to crit an unarmoured target with this, and it has perfect accuracy unlike the L6, meaning from even short distances this crits faster. Now, on top of that, it comes with explosive rounds that can crit instantly in one shot, and do collateral damage to those nearby. This is extraordinarily overpowered. Also the .50 anti-material ammunition box isn't marked as contraband.

**Comment by whateverusername0** [comment]
OP nerf

**Comment by a-person5660** [comment]
> From my testing, it takes ~1 second to crit an unarmoured target with this, and it has perfect accuracy unlike the L6, meaning from even short distances this crits faster. Now, on top of that, it comes with explosive rounds that can crit instantly in one shot, and do collateral damage to those nearby. This is extraordinarily overpowered. Also the .50 anti-material ammunition box isn't marked as contraband.

i will change the damage down to 35 peirce which wont change DPS on an unarmoured target but will hurt it against armoured targets, is that ok?
in my testing the explosion kills an unarmoured target in 3 hits, same as the bullet so im not sure whats going on there.

**Comment by a-person5660** [comment]
in the testing video it kills an armoured target in 4 hits, did the math and that means each round does ~40-50 damage to an unarmoured target, gonna test it now.

**Review by whateverusername0 — APPROVED** [review]

**Comment by a-person5660** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

bruh, im not free to fix this for another day or three, can someone else fix it lmao.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml [hunk 5]
```diff
@@ -7,7 +7,7 @@
   - type: Projectile
     damage:
       types:
-        Piercing: 22
+        Piercing: 26
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:10]: Why the hell are you changing this

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 6]
```diff
@@ -485,7 +485,7 @@
   description: uplink-ammo-bundle-desc
   productEntity: ClothingBackpackDuffelSyndicateAmmoFilled
   cost:
-    Telecrystal: 75 # goob
+    Telecrystal: 75
```
Comment (Piras314) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:488]: Why did you remove the # Goobstation comment

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml [hunk 7]
```diff
@@ -30,7 +30,7 @@
     damage:
       types:
         Blunt: 2
-        Heat: 20
+        Heat: 24
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:33]: Why the hell are you changing this

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml [hunk 8]
```diff
@@ -41,6 +41,6 @@
   - type: Projectile
     damage:
       types:
-        Radiation: 10
-        Piercing: 12
+        Radiation: 12
```
Comment (Piras314) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:44]: Why the hell are you changing this

**Review by Piras314 — COMMENTED** [review]
This is china-lake meets L6 saw, it's not a side-grade to the L6 whatsoever, and should be made much more expensive. Also you modify damage values for a ton of stuff you should not be touching in this PR

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 6 above)
Comment (a-person5660) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:488]: its not actually removing anything, i considered upping the price to 90 which is when i added the goob comment, then i was convinced by the goobers in the discord to leave it as is, i dont know why theres even a diff here.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml (hunk 5 above)
Comment (a-person5660) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:10]: i thought it was cringe that .20 is so weak, i guess i should make it a seperate PR.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml (hunk 7 above)
Comment (a-person5660) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:33]: part of the above

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml (hunk 8 above)
Comment (a-person5660) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:44]: part of the above

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 6 above)
Comment (Piras314) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:488]: you're removing the goob comment when there originally was/is a goob comment there that I added because it's obviously not 75tc on wizden (they have a different scale).

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml (hunk 5 above)
Comment (Piras314) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/rifle.yml:10]: yeah, you should

**Comment by Piras314** [comment]
It comes with 5.5 mags if you include explosive + the free bullet box

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 6 above)
Comment (a-person5660) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:488]: oh mb, i must have accidentally removed the comment without realising there was meant to be one when undoing the change.

**Comment by a-person5660** [comment]
> It comes with 5.5 mags if you include explosive + the free bullet box

yea but you gotta actually load the mags, you cant just magdump and if you get caught out without any loaded mags your fucked cause most of your TC is spent on this.

**Comment by a-person5660** [comment]
so now you get 51 rounds of normal rounds and 10 explosive rounds in the bundle. however with only three 10 round mags it means you need to constantly stop to top up those mags. getting an ammo bundle only provides you with two mags and no extra bullets so you will run dry VERY quickly. and with no way to get more explosive bullets except for the ammo bundle (75 TC and has only 10 in it). besides im active in the community and if it ends up being too OP i will make a PR to nerf it appropriately pretty quickly.

**Comment by a-person5660** [comment]
also keep in mind that the ebow does similar damage at a lower fire rate with more utility (2 hit stun + other modes like peirce) and unlimited ammo, it costs 80 TC.

**Comment by Piras314** [comment]
Nukies can't get EBow

**Comment by Piras314** [comment]
Reducing mag size is not at all a good solution to something being OP. You nerf its damage/abilities, or you make it very expensive. I don't know why you refuse to accept the fact that this is a 100% accuracy L6 SAW (thus better than the L6) that can also fire explosives. It is.

**Comment by Piras314** [comment]
sounds good to me. also conflicts

**Comment by a-person5660** [comment]
I don't know how to fix the conflict from my phone but I will reduce fire rate and make it nukie only now.

**Comment by a-person5660** [comment]
idk why test fail

**Comment by a-person5660** [comment]
good to merge :trollface:

**Comment by Aidenkrz** [comment]
Waiting on Admin Review.
